### DISCUSSION TITLE
Octokit Network errors

### DISCUSSION THREAD

**Discussion by edenhermelin** [octokit/octokit.js#2162] [Q&A]
Hi,

I wanted to know if there is a way / best practice to "suppress" the network errors that Octokit might get.

I know that eventually if Octokit makes a request to the GH REST API and it gets rejected (with 404 for example), the network error will be shown in the browser's console. So I wonder if there is a way to "check if exists"/ "check if can fetch data" before actually requesting it.
Because currently, the library causes a lot of errors on the browser and I'd like to make it a little cleaner :)

thanks!

**Comment by gr2m** [comment]
I don't think that the `octokit` package logs error messages in case of a `404`. Which package are you using exactly?

For browsers, I'd recommend the minimal amount of code, which is https://github.com/octokit/core.js. Then add only the plugins you need, such as pagination for REST APIs: https://github.com/octokit/plugin-paginate-rest.js

It sounds like you are using the https://github.com/octokit/plugin-request-log.js, which is built into `@octokit/rest`:
https://github.com/octokit/rest.js/blob/64d738900cfc9f4e13bb7618da9af39723a827f3/src/index.ts#L10

**Comment by edenhermelin** [comment]
Hey @gr2m ,
Thanks for the response!

I'm using `@octokit/rest` library and getting these errors whenever I'm trying to access something that I can't access/doesn't exist in GitHub.
Let's take the following use case:
* I have a function that fetches the content of a file from Github. 
* This function uses `octokit.repos.getContent` with the `filePath` param.
* The provided `filePath` doesn't exist in the repo, I get 404 (which is expected, as the file doesn't exist and it's a 404 on an HTTP request after all.. ),
Getting this network error triggers the browser and it prints immediately the unpleasant error (the code is from the "fetchWrapper" from within the library):
![image](https://user-images.githubusercontent.com/10133343/137804413-a0114ec1-be98-44c4-8828-0841a93e301f.png)

so what I'm trying to do is to avoid getting the 404 errors and decide by myself when to console.error about a missing resource. 
So I guess that if there is an option to check somehow if a resource is accessible before trying+failing to fetch the resource it will be great.
